jwt/src/main/java/me/uport/sdk/jwt/JWTTools.kt

```diff
@@ -251,6 +252,42 @@ class JWTTools(
         throw InvalidJWTException("DID document for ${payload.iss} does not have any matching public keys")
     }
 
+    /**
+     * This method uses the [auth] param to determine how to filter the list of publicKeys and authenticators
+     *
+     */
+    private suspend fun resolveAuthenticator(alg: String, issuer: String, auth: Boolean): List<PublicKeyEntry> {
+
+        if (alg != JwtHeader.ES256K || alg != JwtHeader.ES256K_R)
+            throw InvalidAlgorithmParameterException("No supported signature types for algorithm $alg")
+
+        val doc = UniversalDID.resolve(issuer)
+
+        val authenticationKeys = if (auth) {
+            doc.authentication.map { it.publicKey }
+        } else {
+            listOf<AuthenticationEntry>() // return an empty list
+        }
+
+
+        val authenticators = doc.publicKey.filter {
+
+            // filter public keys which belong to the list of supported key types
+            //TODO: Make supported types a list to allow for more expressive and manageable checking
+            ((it.type == DelegateType.Curve25519EncryptionPublicKey ||
```

Comment:
You can get a list by using `DelegateType.values()` but it's guaranteed to be one of those values anyway because it is enforced during deserialization.

If the intent is to filter by a subset of the supported `DelegateType`s then it is worth defining a `listOf()` some of those values

Reply:
Yeah, I highlighted this in my TODO comment.